models/project_board.go

```diff
@@ -0,0 +1,226 @@
+	switch p {
+	case None, BasicKanban, BugTriage:
+		return true
+	default:
+		return false
+	}
+}
+
+func createBoardsForProjectsType(sess *xorm.Session, project *Project) error {
+
+	var items []string
+
+	switch project.BoardType {
+
+	case BugTriage:
+		items = setting.Repository.ProjectBoardBugTriageType
+
+	case BasicKanban:
+		items = setting.Repository.ProjectBoardBasicKanbanType
+
+	case None:
+		fallthrough
+	default:
+		return nil
+	}
+
+	if len(items) == 0 {
+		return nil
+	}
+
+	var boards = make([]ProjectBoard, 0, len(items))
+
+	for _, v := range items {
+		boards = append(boards, ProjectBoard{
+			CreatedUnix: timeutil.TimeStampNow(),
+			UpdatedUnix: timeutil.TimeStampNow(),
+			CreatorID:   project.CreatorID,
+			Title:       v,
+			ProjectID:   project.ID,
+			RepoID:      project.RepoID,
+		})
+	}
+
+	_, err := sess.Insert(boards)
+	return err
+}
+
+// NewProjectBoard adds a new project board to a given project
+func NewProjectBoard(board *ProjectBoard) error {
+	_, err := x.Insert(board)
+	return err
+}
+
+// DeleteProjectBoardByID removes all issues references to the project board.
+func DeleteProjectBoardByID(repoID, projectID, boardID int64) error {
+	sess := x.NewSession()
+	defer sess.Close()
+	if err := sess.Begin(); err != nil {
+		return err
+	}
+
+	if err := deleteProjectBoardByID(sess, repoID, projectID, boardID); err != nil {
+		return err
+	}
+
+	return sess.Commit()
+}
+
+func deleteProjectBoardByID(e Engine, repoID, projectID, boardID int64) error {
+	board, err := getProjectBoard(e, repoID, projectID, boardID)
+	if err != nil {
+		if IsErrProjectBoardNotExist(err) {
+			return nil
+		}
+
+		return err
+	}
+
+	if err = board.removeIssues(e); err != nil {
+		return err
+	}
+
+	if _, err := e.ID(board.ID).Delete(board); err != nil {
+		return err
+	}
+	return nil
+}
+
+// GetProjectBoard fetches the current board of a project
+func GetProjectBoard(repoID, projectID, boardID int64) (*ProjectBoard, error) {
+	return getProjectBoard(x, repoID, projectID, boardID)
+}
+
+func getProjectBoard(e Engine, repoID, projectID, boardID int64) (*ProjectBoard, error) {
+	board := &ProjectBoard{
+		ID:        boardID,
+		RepoID:    repoID,
+		ProjectID: projectID,
+	}
+
+	has, err := e.Get(board)
+	if err != nil {
+		return nil, err
+	} else if !has {
+		return nil, ErrProjectBoardNotExist{RepoID: repoID, BoardID: boardID, ProjectID: projectID}
+	}
+
+	return board, nil
+}
+
+// UpdateProjectBoard updates the title of a project board
+func UpdateProjectBoard(board *ProjectBoard) error {
+	return updateProjectBoard(x, board)
+}
+
+func updateProjectBoard(e Engine, board *ProjectBoard) error {
+	board.UpdatedUnix = timeutil.TimeStampNow()
+	_, err := e.ID(board.ID).Cols(
+		"title",
+		"default",
+		"updated_unix",
+	).Update(board)
+	return err
+}
+
+// GetProjectBoards fetches all boards related to a project
+func GetProjectBoards(repoID, projectID int64) ([]*ProjectBoard, error) {
```

Comment:
Ditto the above; projectID should be unique within an instance and sufficient to identify boards.